client/src/components/overview/Gallery.jsx

```diff
@@ -1,35 +1,122 @@
+  const photos = selectedStyle.photos;
+  const THUMBNAIL_HEIGHT_PX = 60;
+  const THUMBNAIL_GUTTER_PX = 20;
+  const thumbnailsMaxScrollHeight =
+    THUMBNAIL_HEIGHT_PX * photos.length +
+    THUMBNAIL_GUTTER_PX * (photos.length - 1);
+  const thumbnailScrollIncrement =
+    THUMBNAIL_HEIGHT_PX * 7 +
+    THUMBNAIL_GUTTER_PX * 6;
+
   const [mainImage, setMainImage] = useState(photos[0]);
+  const [mainImageIndex, setMainImageIndex] = useState(0);
+  const [thumbnailScroll, setThumbnailScroll] = useState(0);
+
+  // Effects
+  useEffect(() => {
+    setMainImage(photos[mainImageIndex]);
+  }, [mainImageIndex]);
 
   // Event handlers
+  const handleThumbnailArrowClick = (event) => {
+    const direction = event.currentTarget.dataset.direction;
+    if (direction === 'down' && thumbnailScroll + thumbnailScrollIncrement < thumbnailsMaxScrollHeight) {
+      setThumbnailScroll(thumbnailScroll + thumbnailScrollIncrement);
+    } else if (direction === 'up' && thumbnailScroll - thumbnailScrollIncrement >= 0) {
+      setThumbnailScroll(thumbnailScroll - thumbnailScrollIncrement);
+    }
+  };
+
+  const handleImageArrowClick = (event) => {
+    const direction = event.currentTarget.dataset.direction;
+    if (direction === 'right' && mainImageIndex + 1 < photos.length) {
+      setMainImageIndex(mainImageIndex + 1);
+    } else if (direction === 'left' && mainImageIndex - 1 >= 0) {
+      setMainImageIndex(mainImageIndex - 1);
+    }
+  };
+
   const handleThumbnailClick = (event) => {
     const index = event.currentTarget.dataset.index;
-    setMainImage(selectedStyle.photos[index]);
+    setMainImageIndex(index);
+  };
+
+  const Arrow = ({ direction, clickHandler, classStr, imgSrc }) => {
+    return (
+      <div
+        className={`${classStr} ${direction}`}
+        data-direction={direction}
+        onClick={clickHandler}
+      >
+        <img src={imgSrc}></img>
+      </div>
+    );
   };
 
   return (
     <>
       <div className='gallery-thumbnails'>
-        <GalleryThumbnails
-          selectedThumbnail={mainImage.thumbnail_url}
-          photos={photos}
-          handleThumbnailClick={handleThumbnailClick}
+        <Arrow
```

Comment:
This is really neat - my first time seeing a "component-less" component. I can see that these are just functions being called with an image, style, a direction to track event, and the event handler. 

Very cool.